src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -946,6 +966,66 @@ public void TestConnectionStringEnvVarHasHighestPrecedence()
+        /// when the engine is running in Production mode. The error messages in Production mode is
+        /// very generic to not reveal information about the underlying database objects backing the entity.
+        /// This test runs against a MsSql database. However, generic error messages will be returned in Production
+        /// mode when run against PostgreSql and MySql databases.
+        /// </summary>
+        /// <param name="requestType">Type of REST request</param>
+        /// <param name="requestPath">Endpoint for the REST request</param>
+        /// <param name="expectedErrorMessage">Right error message that should be shown to the end user</param>
+        [DataTestMethod]
+        [TestCategory(TestCategory.MSSQL)]
+        [DataRow(RestMethod.Get, "/api/Book/id/one", null, "Invalid value provided for field: id", DisplayName = "Validates the error message for a GET request with incorrect primary key parameter type on a table in production mode")]
+        [DataRow(RestMethod.Get, "/api/books_view_all/id/one", null, "Invalid value provided for field: id", DisplayName = "Validates the error message for a GET request with incorrect primary key parameter type on a view in production mode")]
+        [DataRow(RestMethod.Get, "/api/GetBook?id=one", REQUEST_BODY_WITH_CORRECT_PARAM_TYPES, "Invalid value provided for field: id", DisplayName = "Validates the error message for a GET request on a stored-procedure with incorrect parameter type in production mode")]
+        [DataRow(RestMethod.Get, "/api/GQLmappings/column1/one", null, "Invalid value provided for field: column1", DisplayName = "Validates the error message for a GET request with incorrect primary key parameter type with alias defined for primary key column on a table in production mode")]
+        [DataRow(RestMethod.Post, "/api/Book", REQUEST_BODY_WITH_INCORRECT_PARAM_TYPES, "Invalid value provided for field: publisher_id", DisplayName = "Validates the error message for a POST request with incorrect parameter type in the request body on a table in production mode")]
+        [DataRow(RestMethod.Put, "/api/Book/id/one", REQUEST_BODY_WITH_CORRECT_PARAM_TYPES, "Invalid value provided for field: id", DisplayName = "Validates the error message for a PUT request with incorrect primary key parameter type on a table in production mode")]
+        [DataRow(RestMethod.Put, "/api/Book/id/1", REQUEST_BODY_WITH_INCORRECT_PARAM_TYPES, "Invalid value provided for field: publisher_id", DisplayName = "Validates the error message for a bad PUT request with incorrect parameter type in the request body on a table in production mode")]
+        [DataRow(RestMethod.Patch, "/api/Book/id/one", REQUEST_BODY_WITH_CORRECT_PARAM_TYPES, "Invalid value provided for field: id", DisplayName = "Validates the error message for a PATCH request with incorrect primary key parameter type on a table in production mode")]
```

Comment:
Can rephrase the display name as - "PATCH request with incorrect PK parameter type on a table in production mode" - since something is incorrect, it is implied that it is a negative test and we would be validating the exception. Again up to you, but just a suggestion. Will be short and we won't be missing out on any information.

Reply:
Even though it is implied that in the incase if incorrect primary key parameter type, we would be validating the exception, I feel writing it out in the display name would make it easier when we are looking at the test report from ADO test pipeline reports. So, atleast, for the moment, I'm going to leave the display names as is.